src/plugins/output_format/common.h

```diff
@@ -0,0 +1,241 @@
+* and redistribute Covered Software under those terms instead of these.   *
+*                                                                         *
+* Source is provided to this software because we believe users have a     *
+* right to know exactly what a program is going to do before they run it. *
+* This also allows you to audit the software for security holes.          *
+*                                                                         *
+* Source code also allows you to port DRAKVUF to new platforms, fix bugs, *
+* and add new features.  You are highly encouraged to submit your changes *
+* on https://github.com/tklengyel/drakvuf, or by other methods.           *
+* By sending these changes, it is understood (unless you specify          *
+* otherwise) that you are offering unlimited, non-exclusive right to      *
+* reuse, modify, and relicense the code.  DRAKVUF will always be          *
+* available Open Source, but this is important because the inability to   *
+* relicense code has caused devastating problems for other Free Software  *
+* projects (such as KDE and NASM).                                        *
+* To specify special license conditions of your contributions, just say   *
+* so when you send them.                                                  *
+*                                                                         *
+* This program is distributed in the hope that it will be useful, but     *
+* WITHOUT ANY WARRANTY; without even the implied warranty of              *
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the DRAKVUF   *
+* license file for more details (it's in a COPYING file included with     *
+* DRAKVUF, and also available from                                        *
+* https://github.com/tklengyel/drakvuf/COPYING)                           *
+*                                                                         *
+***************************************************************************/
+
+#ifndef PLUGINS_OUTPUT_FORMAT_COMMON_H
+#define PLUGINS_OUTPUT_FORMAT_COMMON_H
+
+#include "ostream.h"
+
+#include <libdrakvuf/libdrakvuf.h>
+#include <plugins/private.h>
+
+#include <algorithm>
+#include <functional>
+#include <iomanip>
+#include <ios>
+#include <iostream>
+#include <map>
+#include <optional>
+#include <sstream>
+#include <string>
+#include <tuple>
+#include <utility>
+#include <vector>
+
+struct TimeVal
+{
+    glong tv_sec;
+    glong tv_usec;
+};
+
+template<class Value>
+auto keyval(const char* key, const Value& value)
+{
+    return std::make_pair(key, value);
+}
+
+namespace fmt
+{
+
+template<class T>
+struct __val_holder
+{
+    T value;
+
+    __val_holder(T&& v): value(std::forward<T>(v)) {}
+};
+
+/* numeric value */
+template<class T>
+struct Nval: __val_holder<T>
+{
+    Nval(T&& v): __val_holder<T>(std::forward<T>(v)) {}
+};
+
+template<class T, std::enable_if_t<std::is_integral_v<std::remove_reference_t<T>>, int> = 0>
+auto nval(T&& value)
```

Comment:
I'm pretty sure that because one of them is called `Nval` and the other `nval` it will cause a lot of compile errors. I'd propose to rename `Nval` to `Nval_holder` or something like that (same for every other type).

Reply:
Removed deducing helpers, classes deduce types itself now.